modules/ocf_mirrors/manifests/monitoring.pp

```diff
@@ -8,16 +8,18 @@
     $upstream_protocol = 'http',
     $ensure = 'present',
   ) {
-
+  $local_url = "http://mirrors.ocf.berkeley.edu${local_path}/dists/${dist_to_check}/Release"
+  $upstream_url = "${upstream_protocol}://${upstream_host}${upstream_path}/dists/${dist_to_check}/Release"
   if $ensure == 'present' {
     file { "${project_path}/health":
-      content => template("ocf_mirrors/monitoring/${type}-health.erb"),
-      mode    => '0755';
-    }
-
+      ensure => link,
+      target => "/opt/mirrors/bin/${type}-healthcheck",
+      owner  => mirrors,
+      group  => mirrors,
+    }->
     cron { "${title}-health":
-      command => "${project_path}/health",
-      user    => 'mirrors',
+      command => "${project_path}/health ${local_url} ${upstream_url}",
```

Comment:
I'm not a fan of positional arguments for this since I think it's tough to remember the correct order (although this isn't being run by humans), and makes it less clear what is being passed in to the script. Maybe having required flags or something would be better? I'm a little split on this because I think adding the local and upstream mirrors as flags makes it look like they are optional when they should be required as inputs.

Reply:
I had the exact same feelings as you when changing the script. I was really close to making them required flags. If it makes you feel better, the order doesn't *actually* matter for this script. We can make the positional params more flexible and just check that the same url isn't given twice?

Reply:
The order does currently matter, since it subtracts the times from each other, although if it just took the larger one and subtracted the smaller one that would work ok and then the order wouldn't matter here (and you'd need to change the output so that it doesn't assume the first is local and the second is upstream). I think that might be better, although I don't see the order ever changing in how we call the script or anything. In general, if the order doesn't matter, then I think positional arguments are fine, but otherwise I'd be more inclined to lean towards flags.

Reply:
How about using ```abs``` and then what I said above and we're both happy? For example, using data from our debian mirrors, ```local - upstream```.

```python
str(parse('2017-12-21 04:56:19+00:00') - parse('2017-12-21 07:34:54+00:00'))
'-1 day, 21:21:25'
str(abs(parse('2017-12-21 04:56:19+00:00') - parse('2017-12-21 07:34:54+00:00')))
'2:38:35'
```